### PR TITLE
FOLIO-2169 "JSON Schema migration for module update section" added

### PR DESCRIPTION
**Pull request by TarasSpashchenko**
Added a section which describes the steps needed to validate a version update for a module to check JSON schema migration result.

### CODE REVIEW

**Comment by skoczko** [comment]
Guys, should this be broken up into a PR for Okapi and RMB? I'd like to see this merged or published somewhere ASAP.

**Comment by marcjohnson-kint** [comment]
> Guys, should this be broken up into a PR for Okapi and RMB? I'd like to see this merged or published somewhere ASAP.

Are we sure that this is the guidance we want folks to follow? When I read the draft document, there were lots of unanswered questions around guidance of when and how to perform these changes. Have they all been resolved?

Is Okapi the appropriate place for documenting this generally, or is it a FOLIO idea and should be documented centrally? Is the thinking that because it is performed by the Tenant API, which is triggered by Okapi, then this process should belong there?

File: README.md [hunk 1]
```diff
@@ -2217,6 +2218,151 @@ RMB will return a response to the client as follows:
...
+
+For RMB based modules a DB migration will be done automatically based on the content of the schema.json provided within the module. Indeed there are exceptions, but each such case must be considered separately.
+
+
+#### Recommendations for the creation and maintenance of the schema.json file
+* For each table in the “tables” section provide “fromModuleVersion” value even if this table has not been added in this version and already exists in the database. It will eliminate the generation and running of unnecessary SQL statements for the table. You can provide a value for the first production version for example
+    ```
+      {
+        "tableName": "loan_type",
+        "fromModuleVersion": "mod-inventory-storage-1.0.0",
```
Comment (marcjohnson-kint) [inline README.md]: Does `mod-inventory-storage-1.0.0-SNAPSHOT.265` match this from version?

Comment (TarasSpashchenko) [reply README.md]: fromModuleVersion - this field indicates the version in which the table was created / updated in. The same for scripts. ComparableVersion.java forked from maven is used under the hood for comparison. So  mod-inventory-storage-1.0.0-SNAPSHOT.265 is a valid version number, 

This is a result of a comparison between  mod-inventory-storage-1.0.0-SNAPSHOT.265 and mod-inventory-storage-1.0.0-SNAPSHOT
Display parameters as parsed by Maven (in canonical form) and comparison result:
1. mod-inventory-storage-1.0.0-SNAPSHOT.265 == mod-inventory-storage-1-snapshot.265
   mod-inventory-storage-1.0.0-SNAPSHOT.265 > mod-inventory-storage-1.0.0-SNAPSHOT
2. mod-inventory-storage-1.0.0-SNAPSHOT == mod-inventory-storage-1-snapshot

Comment (marcjohnson-kint) [reply README.md]: I don't think I follow. Maybe an example could help me.

If we have a schema entry that states a `from version` of `mod-inventory-storage-1.0.0`

If this was introduced in the build `mod-inventory-storage-SNAPSHOT.274`

What should happen during the tenant API when the `toVersion` is `mod-inventory-storage-SNAPSHOT.274` on a fresh system?

Will the schema migration that states a `from version` of `mod-inventory-storage-1.0.0` be executed?

Comment (TarasSpashchenko) [reply README.md]: Let’s look at a case
-	The current version of a module enabled for a tenant is mod-inventory-storage-SNAPSHOT.274
-	The version of a module that will be enabled for a tenant (it is provided in the ModuleDescriptor.json) is mod-inventory-storage-SNAPSHOT.275
-	The version specified in the “fromModuleVersion” in schema.json is mod-inventory-storage-1.0.0
The comparison will be done between mod-inventory-storage-1.0.0 and mod-inventory-storage-SNAPSHOT.274 and the result is mod-inventory-storage-1.0.0 > mod-inventory-storage-SNAPSHOT.274 so a script/table will be applied/created or changed.

Another case 
-	The current version of a module enabled for a tenant is mod-inventory-storage-1.0.0-SNAPSHOT.274
-	The version of a module that will be enabled for a tenant (it is provided in the ModuleDescriptor.json) is mod-inventory-storage-1.0.0-SNAPSHOT.275
-	The version specified in the “fromModuleVersion” in schema.json is mod-inventory-storage-1.0.0
The comparison will be done between mod-inventory-storage-1.0.0 and mod-inventory-storage-1.0.0-SNAPSHOT.274 and the result is mod-inventory-storage-1.0.0 > mod-inventory-storage-1.0.0-SNAPSHOT.274 so a script/table will be applied/created or changed as well.

In both cases we have the correct behavior.

**Pull request closed without merging**